packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -835,10 +835,12 @@ export default util.createRule<Options, MessageId>({
       );
 
       if (strings.length) {
-        if (strings.some(type => type.isStringLiteral() && type.value !== '')) {
-          variantTypes.add('truthy string');
-        } else {
+        if (
+          strings.some(type => !type.isStringLiteral() || type.value === '')
+        ) {
           variantTypes.add('string');
+        } else {
+          variantTypes.add('truthy string');
         }
```

Comment:
This also works, but is less efficient. 

```suggestion
        if (
          strings.every(type => type.isStringLiteral() && type.value !== '')
        ) {
          variantTypes.add('truthy string');
        } else {
          variantTypes.add('string');
        }
```

Reply:
@kozlovvski They both have the same efficiency :) They both bail out at the same expected number of iterations.

Reply:
@Josh-Cena didn't know that `every` returns early -> if this is the case, then probably the latter is more readable